src/parser/ast.ts

```diff
@@ -21,46 +21,51 @@ export enum NodeType {
   block_comment = 'block_comment',
 }
 
-export type StatementNode = {
+type BaseNode = {
+  leadingComments?: CommentNode[];
+  trailingComments?: CommentNode[];
+};
+
+export type StatementNode = BaseNode & {
```

Comment:
these should be `export interface XNode extends BaseNode`

Reply:
I've actually been pondering over switching away from using `interface` completely. Everything done with `interface` can also be done with `type`, with one exception:

- declaring two interfaces with same name will result in them being merged together into a single interface. A behavior one almost never wants.

On the other side, there are lots of things one can do with `type` which can't be done with `interface`.

So there's two options:

- use both type and interface (with rules to decide when to use one or the other)
- use only type (with no decisions required over which syntax to use in which case).

Reply:
I've always followed the convention of:
`interface` for all object-like declarations, `type` for everything else (ie. unions, aliases, etc.)

Reply:
I'll switch to interface for now.